libyul/AsmAnalysis.cpp

```diff
@@ -601,8 +601,18 @@ bool AsmAnalyzer::validateInstructions(std::string const& _instructionIdentifier
 	auto const builtin = EVMDialect::strictAssemblyForEVM(EVMVersion{}).builtin(YulString(_instructionIdentifier));
 	if (builtin && builtin->instruction.has_value())
 		return validateInstructions(builtin->instruction.value(), _location);
-	else
-		return false;
+
+	if (EVMDialect::strictAssemblyForEVM(EVMVersion()).reservedIdentifier(YulString(_instructionIdentifier)))
```

Comment:
I'm confused - why is this function EVM specific?

Reply:
Good question.

Reply:
Fixed.